README.md

```diff
@@ -35,60 +35,62 @@ to use any of the available tasks and is encouraged to improve the, develop new
+- **`mask`** - Tasks used for masking of data and calculation of cloud/snow/other masks.
+- **`ml-tools`** - Various tools that can be used before or after the machine learning process.
+- **`visualization`** - Visualization tools for the core elements of eo-learn.
 
 ## Installation
 
-### PyPi distribution
+### Requirements
 
-The package requires Python version **>=3.8** . It can be installed with:
+Before installing `eo-learn` on **Linux** it is recommended to install the following system libraries:
 
 ```bash
-pip install eo-learn
+sudo apt-get install gcc libgdal-dev graphviz proj-bin libproj-dev libspatialindex-dev
 ```
 
-In order to avoid heavy package dependencies it is possible to install each subpackage separately:
+Before installing `eo-learn` on **Windows** it is recommended to install the following packages from [Unofficial Windows wheels repository](https://www.lfd.uci.edu/~gohlke/pythonlibs/):
 
 ```bash
-pip install eo-learn-core
-pip install eo-learn-coregistration
-pip install eo-learn-features
-pip install eo-learn-geometry
-pip install eo-learn-io
-pip install eo-learn-mask
-pip install eo-learn-ml-tools
-pip install eo-learn-visualization
+gdal
+rasterio
+shapely
+fiona
 ```
 
-Before installing `eo-learn` on **Linux** it is recommended to install the following system libraries:
+One of the `[MASK_EXTRA]` dependencies is the `lightgbm` package. On Windows it requires 64 bit Python distribution. If having problems during installation please check [LightGBM installation guide](https://lightgbm.readthedocs.io/en/latest/Installation-Guide.html).
+
+### PyPi distribution
+
+The package requires Python version **>=3.8** . It can be installed with:
 
 ```bash
-sudo apt-get install gcc libgdal-dev graphviz proj-bin libproj-dev libspatialindex-dev
+pip install eo-learn
 ```
 
-Before installing `eo-learn` on **Windows** it is recommended to install the following packages from [Unofficial Windows wheels repository](https://www.lfd.uci.edu/~gohlke/pythonlibs/):
+In order to avoid heavy package dependencies it is possible to install various scopes of package dependencies. Dependencies for each of the subfolders above (or their extra dependencies) can be installed via e.g.:
 
 ```bash
-gdal
-rasterio
-shapely
-fiona
+pip install "eo-learn[FEATURES]"
+pip install "eo-learn[FEATURES_EXTRA]"
 ```
 
-One of dependencies of `eo-learn-mask` subpackage is `lightgbm` package. On Windows it requires 64 bit Python distribution. If having problems during installation please check [LightGBM installation guide](https://lightgbm.readthedocs.io/en/latest/Installation-Guide.html).
+Collections of dependencies are also available, some of them are:
 
-Some subpackages contain extension modules under `extra` subfolder. Those modules typically require additional package dependencies which don't get installed by default.
+- `[ALL]` for installing the basic dependencies without extras
```

Comment:
this subsection might need to be revised after we finalize extras, somethings might change

Reply:
added `[ZARR]` as discussed